File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int left = 0, right = nums.length-1, count = 0;

        while (left <= right) {
            if (nums[left] == val){
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
                count++;
                right--;
            } else {
                left++;
            }
        }

        return nums.length-count;
    }
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        ret = len(nums) - nums.count(val)
        result = [x for x in nums if x!= val]
        for i,_ in enumerate(result):
            nums[i] = result[i]
        return ret
        
